From ef0b62350a3da00ef7b56b6e096c02fb565a3b70 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Wed, 29 Nov 2017 06:14:06 +0100
Subject: [PATCH] moveit_ros_perception: make OpenGL parts optional

But build everything by default.

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>

[Ported Dmitry's patch from indigo to current kinetic branch]

Upstream-Status: Submitted [https://github.com/ros-planning/moveit/pull/698]

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 moveit_ros/perception/CMakeLists.txt | 36 +++++++++++++++++++++---------------
 1 file changed, 21 insertions(+), 15 deletions(-)

diff --git a/moveit_ros/perception/CMakeLists.txt b/moveit_ros/perception/CMakeLists.txt
index 95fbc58..064fb74 100644
--- a/moveit_ros/perception/CMakeLists.txt
+++ b/moveit_ros/perception/CMakeLists.txt
@@ -3,18 +3,25 @@ project(moveit_ros_perception)
 
 add_compile_options(-std=c++11)
 
+option(BUILD_OPENGL "Build the parts that depends on OpenGL" ON)
+
 if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
 find_package(Boost REQUIRED thread signals)
 
-find_package(OpenGL REQUIRED)
-find_package(GLEW REQUIRED)
-find_package(GLUT REQUIRED)
-if (OPENGL_FOUND)
-  set(gl_LIBS ${gl_LIBS} ${OPENGL_LIBRARIES})
-endif(OPENGL_FOUND)
+if(BUILD_OPENGL)
+  find_package(OpenGL REQUIRED)
+  find_package(GLEW REQUIRED)
+  find_package(GLUT REQUIRED)
+  if (OPENGL_FOUND)
+    set(gl_LIBS ${gl_LIBS} ${OPENGL_LIBRARIES})
+  endif(OPENGL_FOUND)
+  set(perception_GL_INCLUDE_DIRS "mesh_filter/include" "depth_image_octomap_updater/include")
+  set(SYSTEM_GL_INCLUDE_DIRS ${GLEW_INCLUDE_DIR} ${GLUT_INCLUDE_DIR})
+endif(BUILD_OPENGL)
+
 if(APPLE)
   find_package(X11 REQUIRED)
 endif(APPLE)
@@ -47,13 +54,12 @@ find_package(OpenCV)
 
 catkin_package(
   INCLUDE_DIRS
-    mesh_filter/include
     lazy_free_space_updater/include
-    depth_image_octomap_updater/include
     point_containment_filter/include
     occupancy_map_monitor/include
     pointcloud_octomap_updater/include
     semantic_world/include
+    ${perception_GL_INCLUDE_DIRS}
     ${OCTOMAP_INCLUDE_DIRS}
   LIBRARIES
     moveit_lazy_free_space_updater
@@ -70,21 +76,19 @@ catkin_package(
     EIGEN3
 )
 
-include_directories(mesh_filter/include
-                    lazy_free_space_updater/include
-                    depth_image_octomap_updater/include
+include_directories(lazy_free_space_updater/include
                     point_containment_filter/include
                     occupancy_map_monitor/include
                     pointcloud_octomap_updater/include
                     semantic_world/include
+                    ${perception_GL_INCLUDE_DIRS}
                     ${Boost_INCLUDE_DIRS}
                     ${catkin_INCLUDE_DIRS}
                     )
 include_directories(SYSTEM
                     ${OpenCV_INCLUDE_DIRS}
                     ${EIGEN3_INCLUDE_DIRS}
-                    ${GLEW_INCLUDE_DIR}
-                    ${GLUT_INCLUDE_DIR}
+                    ${SYSTEM_GL_INCLUDE_DIR}
                     ${X11_INCLUDE_DIR}
                     )
 
@@ -96,8 +100,10 @@ add_subdirectory(lazy_free_space_updater)
 add_subdirectory(point_containment_filter)
 add_subdirectory(occupancy_map_monitor)
 add_subdirectory(pointcloud_octomap_updater)
-add_subdirectory(mesh_filter)
-add_subdirectory(depth_image_octomap_updater)
+if (BUILD_OPENGL)
+  add_subdirectory(mesh_filter)
+  add_subdirectory(depth_image_octomap_updater)
+endif(BUILD_OPENGL)
 
 add_subdirectory(semantic_world)
 
-- 
2.7.4

